samples/node/pub_sub/index.ts

```diff
@@ -0,0 +1,194 @@
+/* Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License").
+* You may not use this file except in compliance with the License.
+* A copy of the License is located at
+*
+*  http://aws.amazon.com/apache2.0
+*
+* or in the "license" file accompanying this file. This file is distributed
+* on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+* express or implied. See the License for the specific language governing
+* permissions and limitations under the License.
+*/
+
+import { mqtt, auth, http, io, iot } from 'aws-crt';
+import { TextDecoder } from 'util';
+
+type Args = { [index: string]: any };
+
+const yargs = require('yargs');
+yargs.command('*', false, (yargs: any) => {
+    yargs
+        .option('endpoint', {
+            alias: 'e',
+            description: "Your AWS IoT custom endpoint, not including a port. "  +
+              "Ex: \"abcd123456wxyz-ats.iot.us-east-1.amazonaws.com\"",
+            type: 'string',
+            required: true  
+        })
+        .option('ca_file', {
+            alias: 'r',
+            description: 'FILE: path to a Root CA certficate file in PEM format.',
+            type: 'string',
+            required: false
+        })
+        .option('cert', {
+            alias: 'c',
+            description: 'FILE: path to a PEM encoded certificate to use with mTLS',
+            type: 'string',
+            required: false
+        })
+        .option('key', {
+            alias: 'k',
+            description: 'FILE: Path to a PEM encoded private key that matches cert.',
+            type: 'string',
+            required: false
+        })
+        .option('client_id', {
+            alias: 'C',
+            description: 'Client ID for MQTT connection.',
+            type: 'string',
+            required: true
```

Comment:
likewise, can you have the default client-id be "samples-client-id"
and have it be non-required? for consistency and default-policy-sharing